exporter/exporter.go

```diff
@@ -38,17 +38,19 @@ type Exporter struct {
 	onceRegisterUpdateAge      sync.Once
 	startup                    time.Time
 	promHandler                http.Handler
+	log                        logrus.Logger
 
 	mu     sync.RWMutex
 	start  time.Time
 	end    time.Time
 	oldEnd time.Time
 }
 
-func NewExporter(c *Config) *Exporter {
+func NewExporter(c *Config, log logrus.Logger) *Exporter {
```

Comment:
This is an API change. Please move dependency injection to a new func like `NewExporterWithLogger` or so.

Also do not hard code a specific implementation (logrus), but accept an interface instead. You may define that interface in this package.

Reply:
Note2: log should be a pointer. This copies the logger and its mutex with it. Linters (staticcheck?) screams at this.